pallets/treasury-buyout-extension/src/lib.rs

```diff
@@ -185,6 +221,58 @@ pub mod pallet {
 			Self::deposit_event(Event::<T>::BuyoutLimitUpdated { limit });
 			Ok(().into())
 		}
+
+		/// Allows root to update the allowed currencies for buyout.
+		///
+		/// Parameters
+		///
+		/// - `origin`: Origin must be root.
+		/// - `updates`: List of updates to be applied to `AllowedCurrencies` storage.
+		///
+		/// Emits `AllowedAssetsForBuyoutUpdated` event when successful.
+		#[pallet::call_index(2)]
+		#[pallet::weight(<T as pallet::Config>::WeightInfo::update_allowed_assets())]
+		#[transactional]
+		pub fn update_allowed_assets(
+			origin: OriginFor<T>,
+			updates: Vec<BuyoutAssetUpdate<CurrencyIdOf<T>>>,
+		) -> DispatchResultWithPostInfo {
+			ensure_root(origin)?;
+
+			// Ensure number of `AllowedCurrencies` storage updates doesn't exceed the maximum allowed
+			ensure!(
+				updates.len() <= T::MaxAllowedCurrencyUpdates::get() as usize,
+				Error::<T>::ExceedsNumberOfAllowedUpdates
+			);
+
+			// Used for ensuring that native token is not allowed for buyout
+			let basic_asset = <T as orml_currencies::Config>::GetNativeCurrencyId::get();
+
+			// Used for the event data
+			let mut added_assets = Vec::new();
+			let mut removed_assets = Vec::new();
+
+			for update in updates.clone() {
+				match update {
+					BuyoutAssetUpdate::Add(asset) => {
+						// Ensure native token is not added to the allowed assets for buyout
+						ensure!(asset != basic_asset, Error::<T>::NativeTokenNotAllowed);
+						AllowedCurrencies::<T>::insert(asset, ());
+						added_assets.push(Some(asset));
+					},
+					BuyoutAssetUpdate::Remove(asset) => {
+						AllowedCurrencies::<T>::remove(asset);
```

Comment:
Does this throw if the asset doesn't exist yet?

Reply:
No, it doesn't throw any error if the asset doesn't exist. However, your suggestion would simplify the logic so I will make the list replaceable, ignoring the last state 👍🏼

Reply:
Done 👍🏼